services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_core.py

```diff
@@ -1135,41 +1148,34 @@ async def _autoscale_cluster(
     # 3. start buffer instances to cover the remaining tasks
     cluster = await _start_buffer_instances(app, cluster, auto_scaling_mode)
 
+    # 4. scale down unused machines
+    cluster = await _scale_down_unused_cluster_machines(app, cluster, auto_scaling_mode)
```

Comment:
LEt's see if i get it right.

- 1. So a key change i see is that you always try to scale down BEFORE checking whether you can scale up instead of as *AN ALTERNATIVE* to it. 
- 2. `_scale_down_unused_cluster_machines` basically wraps the same calls you have before in one call
  - 1. `_try_scale_down_cluster` restricts further the instances corresponding to `drained_nodes` (i.e. those in "terminating" are still not considered)
  

I believe 1. is the key change that solves this issue. right?

Reply:
correct.
Not sure what you mean with terminating instances are still not considered. 

When the scale down process starts it goes like so:
1. waits a while to see if the instance is really empty
2. drains the instance by changing a label on it OR docker drain
3. waits a while to see if nothing happens
4. starts the termination process, basically removes it from the usable instances
5. waits a while
6. terminates the instance

Reply:
so if you mean that terminating instances are still counted as running instances, yes that is true. You currently have to wait for them to be really terminated before you can get another one. that is fair no?

Reply:
so if you mean that terminating instances are still counted as running instances, yes that is true. You currently have to wait for them to be really terminated before you can get another one. that is fair no?